Completed array manipulation-2 exercise

sanfordhuynh opened the pull request:
![array manipulation2](https://github.com/sanfordhuynh/m1-0723-code-solutions/assets/16739580/2944bd18-d018-4b64-add6-43c48385e064)

sanfordhuynh committed:
Completed array manipulation-2 exercise

kurtyazdizadeh commented on array-manipulation-2/src/chunk.js:
```diff
@@ -0,0 +1,27 @@
+    return [];
+  }
+
+  let currentChunk = [];
```
reference vars should use `const`

kurtyazdizadeh commented on array-manipulation-2/src/chunk.js:
```diff
@@ -0,0 +1,27 @@
+    // check if the current chunk is at the desired size or if at the end of array
+    if (currentChunk.length === size || i === array.length - 1) {
+      newArray.push(currentChunk);
+      currentChunk = []; // reset for the next chunk
```
if setting the reference var with `const`, another way to reset the array is to do `currentChunk.length = 0;`

kurtyazdizadeh commented on array-manipulation-2/src/drop.js:
```diff
@@ -0,0 +1,8 @@
+    return [];
+  }
+
+  return array.slice(count);
```
can you solve this without using the `slice` method?

kurtyazdizadeh reviewed (changes requested):
Please review my comments, if you have any questions let me know!

sanfordhuynh committed:
Tried to change let currentChunk to const and make it currentChunk.length = 0 but couldn't get it to work since currentChunk is modified to 0 each time.

Redone the drop method without using slice() and have added pseudocode to the exercise

kurtyazdizadeh reviewed (approved):
I see what you meant by trying to change it. Disregard my comment then, it works the way you had it, there are rare occasions you can use `let` and this looks like one of them!

Approved 👍
